File: GFG/Array/Day2/BuyAndSellStock.java
public class BuyAndSellStock {
    public int maxProfit(int[] prices) {
        // Code here
        int start,end,profit,curprofit;
        start=0;
        end=1;
        profit=0;
        
        while(end<prices.length){
            if(prices[start]>prices[end]){
                start=end;
            }
            curprofit=prices[end]-prices[start];
            profit=Math.max(profit,curprofit);
            end+=1;
        }
        
        return profit;
    }
}

File: GFG/Array/Day2/BuyAndSellStock.py
class Solution:
    def maxProfit(self, prices):
        # code here
        start=0
        end=1
        profit=0
        curProfit=0
        
        while(end<len(prices)):
            if(prices[start]>prices[end]):
                start=end
            curProfit=prices[end]-prices[start]
            profit=max(profit,curProfit)
            end+=1
            
        return profit